### PR TITLE
MOSIP-45275 - Automated signup not automated test cases

### PR DESCRIPTION
**Pull request by prathmeshj12**
MOSIP-45275 - Automated signup not automated test cases

## Summary by CodeRabbit

* **New Features**
  * Added coverage for different-identifier, expired transaction, consent, profile-update, trust-framework, and claim-availability signup scenarios.
  * Added positive and negative WebSocket tests, including large frames and invalid message details.
  * Added validation for redirect ID-token results and identity-verifier responses.
  * Added mock and non-mock claim-status response variations.

* **Bug Fixes**
  * Improved WebSocket outcome validation and corrected WebSocket test integration.
  * Added checks for invalid tokens, expired transactions, and expired authorization hints.

### CODE REVIEW

File: api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/SimplePostForAutoGenId.java [hunk 1]
```diff
@@ -183,11 +183,28 @@ public void test(TestCaseDTO testCaseDTO)
...
+				String actualJson = response.asString();
+				try {
+					String idToken = new JSONObject(actualJson).getJSONObject(GlobalConstants.RESPONSE)
+							.getString(idKeyName);
+					actualJson = AdminTestUtil.decodeAndCombineJwt(idToken);
+				} catch (Exception e) {
+					logger.info(
+							"idToken not decodable, falling back to raw response validation (likely a negative test case): "
+									+ e.getMessage());
+				}
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/SimplePostForAutoGenId.java:196]: Fixed — now logs `e.getClass().getSimpleName()` instead of `e.getMessage()`, so no
token material can leak while the fallback path keeps its diagnostic value.

Heads-up: the committable suggestion drops the `+` before `e.getClass()` and won't
compile. Applied with the operator restored.

File: api-test/src/main/resources/signup/AddIdentity/AddIdentity.yml [hunk 2]
```diff
@@ -308,4 +308,60 @@ AddIdentity:
...
+  "phone": "$PHONENUMBERFORIDENTITY$",
+  "referenceIdentityNumber": "6789545678878",
+  "version": "v1",
+   "introducerRID": "212124324784879",
+   "introducerUIN": "212124324784879",
+   "category": "individualBiometrics",
+   "requesttime": "$TIMESTAMP$"
+}'
+      output: '{
+  "status":"ACTIVATED"
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/AddIdentity/AddIdentity.yml:366]: False positive — all six files already end with `}'`. It's on the last line with no
trailing newline, which renders as `\ No newline at end of file` in the diff.

Verified: all six parse cleanly with a YAML loader. Appending a second `}'` would
introduce the very syntax error being flagged. (46 of 60 YAMLs here have no trailing
newline — existing norm.)

File: api-test/src/main/resources/signup/ClaimDetailsConsentNoCapture/AuthenticateConsentNoCapture.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        "sendOtp":{
+    "encodedHash": "$ID:ESignet_OAuthDetailsRequest_V3_ConsentNoCapture_uin_SameClaims_Smoke_sid_encodedResp$",
+    "requestTime": "$TIMESTAMP$",
+    "transactionId": "$ID:ESignet_OAuthDetailsRequest_V3_ConsentNoCapture_uin_SameClaims_Smoke_sid_transactionId$",
+    "individualId": "$ID:AddIdentity_L2_Valid_Parameters_smoke_Pos_UIN$",
+    "otpChannels": [{channel: "email"},{channel: "phone"}],
+    "sendOtpReqTemplate": "signup/SendOtp/SendOtp",
+    "sendOtpEndPoint": "/v1/esignet/authorization/send-otp"
+    }
+    }'
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/ClaimDetailsConsentNoCapture/AuthenticateConsentNoCapture.yml:29]: Applied — `[{"channel": "email"},{"channel": "phone"}]`.

File: api-test/src/main/resources/signup/ClaimDetailsProfileUpdateVoluntary/AuthenticateProfileUpdateVoluntary.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      	"transactionId": "$ID:ESignet_OAuthDetailsRequest_V3_ProfileUpdateVoluntary_uin_Smoke_sid_transactionId$",
+      	"individualId": "$ID:AddIdentity_L2_Valid_Parameters_smoke_Pos_UIN$",
+      	"authFactorType" : "OTP",
+      	"challenge" : "$ID:AddIdentity_L2_Valid_Parameters_smoke_Pos_PHONE$@phone",
+        "sendOtp":{
+    "encodedHash": "$ID:ESignet_OAuthDetailsRequest_V3_ProfileUpdateVoluntary_uin_Smoke_sid_encodedResp$",
+    "requestTime": "$TIMESTAMP$",
+    "transactionId": "$ID:ESignet_OAuthDetailsRequest_V3_ProfileUpdateVoluntary_uin_Smoke_sid_transactionId$",
+    "individualId": "$ID:AddIdentity_L2_Valid_Parameters_smoke_Pos_UIN$",
+    "otpChannels": [{channel: "email"},{channel: "phone"}],
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/ClaimDetailsProfileUpdateVoluntary/AuthenticateProfileUpdateVoluntary.yml]: Applied — `[{"channel": "email"},{"channel": "phone"}]`.\

File: api-test/src/main/resources/signup/ClaimDetailsTrustFrameworkMismatch/AuthenticateTrustFrameworkMismatch.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+AuthenticateTrustFrameworkMismatch:
+   Signup_ESignet_AuthenticateTrustFrameworkMismatch_uin_Otp_Smoke:
+      endPoint: /v1/esignet/authorization/v3/authenticate
+      uniqueIdentifier: TC_Signup_ESignet_AuthenticateTrustFrameworkMismatch_01
+      description: Verify authentication succeeds for the trust-framework-mismatch scenario - expected, authentication succeeds
+      role: resident
+      restMethod: post
+      checkErrorsOnlyInResponse: true
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/ClaimDetailsTrustFrameworkMismatch/AuthenticateTrustFrameworkMismatch.yml:8]: Not applying — this would break 2 of the 3 tests.

The flag doesn't skip payload validation. In `OutputValidationUtil` (L470-483)
`compareActuExpValue` always runs the full comparison; the flag is only read in the
`catch (SkipException)`, and that's thrown only when `comparisonDone == false` (L234)
— i.e. nothing was compared. A mismatched/missing field sets FAIL (L217-223)
regardless. So it can't mask a mismatch.

- **Authenticate…** — template `AuthenticateUserResult.hbs` is literally `{}` → empty
  expected map → removing the flag turns PASS into SKIP, cascading to every test using
  `$ID:..._transactionId$`.
- **OAuthDetails…** — no `outputTemplate`, `output: '{ }'` → same.
- **ClaimDetails…** — template does assert `available:true`/`verified:false`, so the
  flag is already a no-op; full validation runs today.

Also existing convention: pre-existing `AuthenticateUserV3/AuthenticateUser.yml` uses
the same flag + same empty template and passes CI.

File: api-test/src/main/resources/signup/ClaimDetailsTrustFrameworkMismatch/AuthenticateTrustFrameworkMismatch.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      	"transactionId": "$ID:ESignet_OAuthDetailsRequest_V3_TrustFrameworkMismatch_uin_Smoke_sid_transactionId$",
+      	"individualId": "$ID:AddIdentity_L2_Valid_Parameters_smoke_Pos_UIN$",
+      	"authFactorType" : "OTP",
+      	"challenge" : "$ID:AddIdentity_L2_Valid_Parameters_smoke_Pos_PHONE$@phone",
+        "sendOtp":{
+    "encodedHash": "$ID:ESignet_OAuthDetailsRequest_V3_TrustFrameworkMismatch_uin_Smoke_sid_encodedResp$",
+    "requestTime": "$TIMESTAMP$",
+    "transactionId": "$ID:ESignet_OAuthDetailsRequest_V3_TrustFrameworkMismatch_uin_Smoke_sid_transactionId$",
+    "individualId": "$ID:AddIdentity_L2_Valid_Parameters_smoke_Pos_UIN$",
+    "otpChannels": [{channel: "email"},{channel: "phone"}],
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/ClaimDetailsTrustFrameworkMismatch/AuthenticateTrustFrameworkMismatch.yml]: Applied — `[{"channel": "email"},{"channel": "phone"}]`.

File: api-test/src/main/resources/signup/ClaimDetailsConsentNoCapture/ClaimDetailsConsentNoCapture.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+ClaimDetailsConsentNoCapture:
+   Signup_ESignet_ClaimDetailsConsentNoCapture_AuthToken_Xsrf_uin_SameClaims_Smoke:
+      endPoint: /v1/esignet/authorization/claim-details
+      uniqueIdentifier: TC_Signup_ESignet_ClaimDetailsConsentNoCapture_01
+      description: Verify claim-details returns consentAction as NOCAPTURE when consent was already captured for the same claims - expected, consentAction=NOCAPTURE
+      role: resident
+      restMethod: get
+      checkErrorsOnlyInResponse: true
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/signup/ClaimDetailsConsentNoCapture/ClaimDetailsConsentNoCapture.yml]: checkErrorsOnlyInResponse should be removed here

File: api-test/src/main/resources/signup/ClaimDetailsNegTC/ClaimDetailsNegTC.yml [hunk 8]
```diff
@@ -265,4 +265,27 @@ ClaimDetailsNegTC:
...
     }
   ]
+}'
+     Signup_ESignet_ClaimDetailsNegTC_Expired_Transaction_Neg:
+      endPoint: /v1/esignet/authorization/claim-details
+      uniqueIdentifier: TC_ESignet_ClaimDetailsNegTC_V3_13
+      description: Verify claim-details is rejected when the transaction is no longer valid - expected, invalid_transaction error
+      role: resident
+      restMethod: get
+      checkErrorsOnlyInResponse: true
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/signup/ClaimDetailsNegTC/ClaimDetailsNegTC.yml]: checkErrorsOnlyInResponse has to be removed here

File: api-test/src/main/resources/signup/IDTAuthenticationNegTC/IDTAuthenticationNegTC.yml [hunk 9]
```diff
@@ -1158,4 +1158,91 @@ IDTAuthenticationNegTC:
...
+    ]
+}'
+   Signup_ESignet_IDTAuthenticationNegTC_AuthToken_Xsrf_Genuinely_Expired_IdToken_Neg:
+      endPoint: /v1/esignet/authorization/v3/authenticate
+      uniqueIdentifier: TC_ESignet_IDTAuthenticationNegTC_45
+      description: Verify IDT authentication is rejected when the id_token has expired - expected, invalid_token or auth_failed error
+      role: resident
+      restMethod: post
+      allowedErrorCodes: invalid_token,auth_failed
+      checkErrorsOnlyInResponse: true
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/signup/IDTAuthenticationNegTC/IDTAuthenticationNegTC.yml]: checkErrorsOnlyInResponse has to be removed

File: api-test/src/main/resources/signup/InitiateIdVerification/InitiateIdVerification.yml [hunk 10]
```diff
@@ -99,5 +99,22 @@ InitiateIdVerification:
       	"state": "urlInfo1724138417665"
 }'
       output: '{
-       
+
+}'
+   Signup_ESignet_InitiateIdVerification_STransId_AuthToken_Xsrf_uin_all_Valid_forGetSlot_differentIdentifier_Smoke_Sid:
+      endPoint: /v1/signup/identity-verification/initiate
+      uniqueIdentifier: TC_ESignet_InitiateIdVerification_07
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/signup/InitiateIdVerification/InitiateIdVerification.yml:142]: can we do output validation for this TC

File: api-test/src/main/resources/signup/PrepareSignupRedirectNegTC/PrepareSignupRedirectNegTC.yml [hunk 11]
```diff
@@ -431,4 +431,26 @@ PrepareSignupRedirectNegTC:
     }'
       output: '{
         "status": "403"
-	}'
\ No newline at end of file
+	}'
+   Signup_ESignet_PrepareSignupRedirectNegTC_AuthToken_Xsrf_STransId_uin_Otp_Expired_Transaction_Neg_sid:
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/signup/PrepareSignupRedirectNegTC/PrepareSignupRedirectNegTC.yml]: for neg TC, sid is not possible

**Review by mohanachandran-s — CHANGES_REQUESTED** [review]

File: api-test/src/main/resources/signup/ClaimDetailsConsentNoCapture/ClaimDetailsConsentNoCapture.yml (hunk 7 above)
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/ClaimDetailsConsentNoCapture/ClaimDetailsConsentNoCapture.yml]: Removed checkErrorsOnlyInResponse → full validation of consentAction: NOCAPTURE

File: api-test/src/main/resources/signup/ClaimDetailsNegTC/ClaimDetailsNegTC.yml (hunk 8 above)
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/ClaimDetailsNegTC/ClaimDetailsNegTC.yml]: Done. I also applied the same across all the negative TCs in this file that assert a specific error payload, so the full error response is validated (kept it only on the valid-smoke case with empty output).

File: api-test/src/main/resources/signup/IDTAuthenticationNegTC/IDTAuthenticationNegTC.yml (hunk 9 above)
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/IDTAuthenticationNegTC/IDTAuthenticationNegTC.yml]: Done — removed on this TC. Note it still uses allowedErrorCodes: invalid_token,auth_failed since the exact code can vary.

File: api-test/src/main/resources/signup/InitiateIdVerification/InitiateIdVerification.yml (hunk 10 above)
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/InitiateIdVerification/InitiateIdVerification.yml:142]: Added output validation — the result template now asserts the identityVerifiers list returned on initiation (processType/active) instead of only checking for errors. Applied it to all the positive initiate TCs as well.

File: api-test/src/main/resources/signup/PrepareSignupRedirectNegTC/PrepareSignupRedirectNegTC.yml (hunk 11 above)
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/PrepareSignupRedirectNegTC/PrepareSignupRedirectNegTC.yml]: Renamed the TC to drop the _sid suffix (..._Expired_Transaction_Neg), since a failing negative TC doesn't produce a valid session. Nothing downstream referenced it.

File: api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/WebScocketConnection.java [hunk 12]
```diff
@@ -122,6 +127,15 @@ public void test(TestCaseDTO testCaseDTO)
...
 		webSocketClient.sendMessage(message1);
 
+		// Data-driven assertion path: a test declares its expected websocket outcome in its output
+		// (expectConnectionFailure and/or expectedFeedbackCodes). One logic handles every negative
+		// and edge case; positive full-flow tests declare neither and fall through to the frame loop.
+		JSONObject expectedOutcome = new JSONObject(testCaseDTO.getOutput());
+		if (expectedOutcome.has("expectedFeedbackCodes") || expectedOutcome.has("expectConnectionFailure")) {
+			assertWebSocketOutcome(webSocketClient, slotId, tempUrl, expectedOutcome);
+			return;
+		}
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/java/io/mosip/testrig/apirig/signup/testscripts/WebScocketConnection.java]: Done — added a safe fallback: String outputStr = testCaseDTO.getOutput(); and default it to "{}" when null/empty before constructing the JSONObject. Positive-flow tests that declare no expected outcome now fall through to the frame loop instead of crashing.

File: api-test/src/main/resources/signup/AuthenticateUserV3/AuthenticateUser.yml [hunk 13]
```diff
@@ -286,4 +286,165 @@ AuthenticateUserV3:
...
+        "transactionId": "$ID:ESignet_OAuthDetailsRequest_V3_AuthToken_Xsrf_uin_all_Valid_forGetSlot_forWsFrameEdge_Smoke_sid_transactionId$",
+        "individualId": "$ID:AddIdentity_L2_Valid_forGetSlot_differentIdentifier_Parameters_smoke_Pos_UIN$",
+        "authFactorType" : "OTP",
+        "challenge" : "$ID:AddIdentity_L2_Valid_forGetSlot_differentIdentifier_Parameters_smoke_Pos_PHONE$@phone",
+        "sendOtp":{
+    "encodedHash": "$ID:ESignet_OAuthDetailsRequest_V3_AuthToken_Xsrf_uin_all_Valid_forGetSlot_forWsFrameEdge_Smoke_sid_encodedResp$",
+    "requestTime": "$TIMESTAMP$",
+    "transactionId": "$ID:ESignet_OAuthDetailsRequest_V3_AuthToken_Xsrf_uin_all_Valid_forGetSlot_forWsFrameEdge_Smoke_sid_transactionId$",
+    "individualId": "$ID:AddIdentity_L2_Valid_forGetSlot_differentIdentifier_Parameters_smoke_Pos_UIN$",
+    "otpChannels": [{channel: "email"},{channel: "phone"}],
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/src/main/resources/signup/AuthenticateUserV3/AuthenticateUser.yml]: Fixed — quoted all channel keys ([{"channel":"email"},{"channel":"phone"}]) across every occurrence in this file. Also applied the same fix to SignupAuthorizeCode/SignupAuthorizeCode.yml, which had the same pattern. Verified all inputs now parse as valid JSON.

File: api-test/testNgXmlFiles/signupSuite.xml [hunk 14]
```diff
@@ -436,6 +547,14 @@
 				name="io.mosip.testrig.apirig.signup.testscripts.SimplePostForAutoGenId" />
 		</classes>
 	</test>
+	<test name="WebSocketConnectionNegTC">
+		<parameter name="ymlFile"
+			value="signup/WebSocketConnectionNegTC/WebSocketConnectionNegTC.yml" />
+		<classes>
+			<class
+				name="io.mosip.testrig.apirig.signup.testscripts.WebScocketConnection" />
```
[the remark this answers is not in the document]
Comment (prathmeshj12) [reply api-test/testNgXmlFiles/signupSuite.xml]: Done — renamed the class WebScocketConnection → WebSocketConnection (file renamed via git mv to preserve history), updated the logger reference, and updated all three references in signupSuite.xml. Also corrected the mention in the module README.

**Review by mohanachandran-s — APPROVED** [review]

**Review by zesu22 — APPROVED** [review]
approving because @mohanachandran-s  has already approved it
